File: hellovg.c
//
// first OpenVG program
// Anthony Starks (ajstarks@gmail.com)
//
// compile and run on the Raspberry Pi:
// cc -Wall -I/opt/vc/include -I/opt/vc/include/interface/vcos/pthreads -c libshapes.c
// cc -Wall -I/opt/vc/include -I/opt/vc/include/interface/vcos/pthreads -c oglinit.c
// cc -Wall -I/opt/vc/include -I/opt/vc/include/interface/vcos/pthreads -o hellovg hellovg.c libshapes.o oglinit.o -L/opt/vc/lib -lGLESv2 -ljpeg && ./hellovg
//
//
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

#include "VG/openvg.h"
#include "VG/vgu.h"
#include "fontinfo.h"
#include "shapes.h"

int main() {
	int width, height;
	init(&width, &height);     // OpenGL, etc initialization

	Start(width, height);                     // Start the picture
	Background(0,0,0);                        // Black background
	Fill(44,77,232,1);                        // Big blue marble
	Circle(width/2, 0, width);                // The "world"
	Fill(255,255,255,1);                      // White text
	TextMiddle(width/2, height/2, 
	"hello, world", SerifTypeface, width/10); // Greetings 
	End();                                    // End the picture

	while (getchar() != '\n')  // look at the pic, end with [RETURN]
		;

	finish();                 // Graphics cleanup
	exit(0);
}

File: go-client/hellovg.go
// first OpenVG program
package main

import (
	"bufio"
	"github.com/ajstarks/openvg"
	"os"
)

func main() {
	width, height := openvg.Init() // OpenGL, etc initialization

	w2 := float64(width / 2)
	h2 := float64(height / 2)
	w := float64(width)

	openvg.Start(width, height)                               // Start the picture
	openvg.BackgroundColor("black")                           // Black background
	openvg.FillRGB(44, 77, 232, 1)                               // Big blue marble
	openvg.Circle(w2, 0, w)                                   // The "world"
	openvg.FillColor("white")                                 // White text
	openvg.TextMid(w2, h2, "hello, world", "serif", width/10) // Greetings 
	openvg.End()                                              // End the picture
	bufio.NewReader(os.Stdin).ReadBytes('\n')                 // Pause until [RETURN]
	openvg.Finish()                                           // Graphics cleanup
}
